huaweicloud/services/dataarts/resource_huaweicloud_dataarts_factory_job.go

```diff
@@ -0,0 +1,1328 @@
+	if rawArray, ok := rawParams.([]interface{}); ok {
+		if len(rawArray) == 0 {
+			return nil
+		}
+		raw, ok := rawArray[0].(map[string]interface{})
+		if !ok {
+			return nil
+		}
+
+		params := map[string]interface{}{
+			"eventType":  utils.ValueIngoreEmpty(raw["event_type"]),
+			"channel":    utils.ValueIngoreEmpty(raw["channel"]),
+			"failPolicy": utils.ValueIngoreEmpty(raw["fail_policy"]),
+			"concurrent": utils.ValueIngoreEmpty(raw["concurrent"]),
+			"readPolicy": utils.ValueIngoreEmpty(raw["read_policy"]),
+		}
+		return params
+	}
+	return nil
+}
+
+func buildCreateJobRequestBodyParam(rawParams interface{}) []map[string]interface{} {
+	if rawArray, ok := rawParams.([]interface{}); ok {
+		if len(rawArray) == 0 {
+			return nil
+		}
+
+		rst := make([]map[string]interface{}, len(rawArray))
+		for i, v := range rawArray {
+			raw := v.(map[string]interface{})
+			rst[i] = map[string]interface{}{
+				"name":  utils.ValueIngoreEmpty(raw["name"]),
+				"value": utils.ValueIngoreEmpty(raw["value"]),
+				"type":  utils.ValueIngoreEmpty(raw["type"]),
+			}
+		}
+		return rst
+	}
+	return nil
+}
+
+func buildCreateJobRequestBodyBasicConfig(rawParams interface{}) map[string]interface{} {
+	if rawArray, ok := rawParams.([]interface{}); ok {
+		if len(rawArray) == 0 {
+			return nil
+		}
+		raw, ok := rawArray[0].(map[string]interface{})
+		if !ok {
+			return nil
+		}
+
+		params := map[string]interface{}{
+			"owner":           utils.ValueIngoreEmpty(raw["owner"]),
+			"priority":        utils.ValueIngoreEmpty(raw["priority"]),
+			"executeUser":     utils.ValueIngoreEmpty(raw["execute_user"]),
+			"instanceTimeout": utils.ValueIngoreEmpty(raw["instance_timeout"]),
+			"customFields":    utils.ValueIngoreEmpty(raw["custom_fields"]),
+		}
+		return params
+	}
+	return nil
+}
+
+func resourceFactoryJobRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getJob: Query the app
+	var (
+		getJobHttpUrl = "v1/{project_id}/jobs/{id}"
+		getJobProduct = "dataarts-dlf"
+	)
+	getJobClient, err := cfg.NewServiceClient(getJobProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts client: %s", err)
+	}
+
+	getJobPath := getJobClient.Endpoint + getJobHttpUrl
+	getJobPath = strings.ReplaceAll(getJobPath, "{project_id}", getJobClient.ProjectID)
+	getJobPath = strings.ReplaceAll(getJobPath, "{id}", d.Id())
+
+	getJobOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+
+	if v, ok := d.GetOk("workspace_id"); ok {
+		getJobOpt.MoreHeaders["workspace"] = v.(string)
+	}
+
+	getJobResp, err := getJobClient.Request("GET", getJobPath, &getJobOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Job")
```

Comment:
it seems that we should convert the error code when job is not exist.
The status is 400, and the response is as following:
```json
{
	"error_code":"DLF.0100",
	"error_msg":"Job[nnnnn] does not exist"
}
```

Reply:
added